modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -496,7 +496,7 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
                 bool keepdims = layerParams.get<int>("keepdims");
                 MatShape targetShape = inpShape;
                 for (int i = 0; i < axes.size(); i++) {
-                    int axis = clamp(axes.get<int>(i), inpShape.size());
+                    int axis = clamp(axes.get<int>(0), inpShape.size());
```

Comment:
This solution is slightly incorrect, it doesn't work with **keepdims**. I think it's better to try something like:
```
MatShape targetShape;
std::vector<bool> shouldDelete(inpShape.size(), false);
for (int i = 0; i < axes.size(); i++) {
    int axis = clamp(axes.get<int>(i), inpShape.size());
    shouldDelete[axis] = true;
}
for (int axis = 0; axis < inpShape.size(); ++axis){
    if (!shouldDelete[axis])
        targetShape.push_back(inpShape[axis]);
    else if (keepdims)
        targetShape.push_back(1);
}
```

Reply:
This way it doesn't brake the tests

Reply:
@sl-sergei Would you like me to make the changes, test, and then commit them to this pull request?

Reply:
> @sl-sergei Would you like me to make the changes, test, and then commit them to this pull request?

@joegeisbauer It was your initiative to make this contribution, and you did great work highlighting the problem, reproducing and proposing a fix. I think you should finish it:)